apps/system/js/base_module.js

```diff
@@ -107,24 +139,48 @@
   var SubmoduleMixin = {
     loadWhenIdle: function(modules) {
       return this.service.request('schedule', () => {
+        var submodules = this._convertModulesToArray(modules || []);
         this.constructor.SUB_MODULES =
-          this.constructor.SUB_MODULES.concat(modules);
+          this.constructor.SUB_MODULES.concat(submodules);
         return this._startSubModules();
       });
     },
+
+    _convertModulesToArray: function(submodules) {
+      var modules = [];
+      var defaulModules;
+      var deviceModules;
+
+      if (Array.isArray(submodules)) {
```

Comment:
How about also testing whether `submodules` is `undefined` here so we can get rid of `xxx || []` from calling this method every time?